RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,213 @@
+
+The steps for human and mouse are more straightforward and are covered in `02-gastric_cancer_tximeta-live.Rmd`. 
+The main difference for using `tximeta` for other species is that we need to set up the reference data in the form of a `linkedTxome` before running `tximeta()` function. 
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love _et al_.
+
+## Libraries and functions
+
+```{r library}
+# Load magrittr for the pipe
+library(magrittr)
+
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+In this instance, we will need to provide `tximeta` with the salmon index for the species we are using. 
+We have prepared this for zebrafish along with other species which you can find more information about [on GitHub](https://github.com/AlexsLemonade/training-txome-prep).
+
+```{r directories}
+# Set up file path to the salmon index for the species we are using
+index_dir <- file.path("index", "Danio_rerio", "short_index")
+```
+
+These directories and files are for pointing to the `quant.sf` files. 
+
+```{r}
+# directory where the data are located
+data_dir <- file.path("data", "zebrafish-cortisol")
+
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "zebrafish-cortisol_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names}
+# Name the output zebrafish-cortisol_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "zebrafish-cortisol_tximeta.RDS")
+```
+
+## Set up reference data for a non-human, non-mouse species 
+
+Here we will set this up for zebrafish, _Danio rerio_.
+You can look at Ensembl's ftp server to see the full list of species and fasta file options: http://ftp.ensembl.org/pub/release-98/fasta/
+
+If you are using a species other than zebrafish, click through the file paths to find the file you want to use for annotation. 
+
+We will store the `base_url` where all the species FASTA files are located and then the second part of the url is species-specific which you may have to replace. 
+
+```{r}
+fasta_base_url <- "ftp://ftp.ensembl.org/pub/release-98/fasta/"
+
+fasta_files <- c(
```

Comment:
I suggest `urls` here to distinguish the fact that these are not files on this machine.

Is it possible to use downloaded files? If so, what is the tradeoff between links and local files?

```suggestion
fasta_urls <- c(
```